music21/meter/base.py

```diff
@@ -489,6 +490,64 @@ def __init__(self, value: str = '4/4', divisions=None):
     def _reprInternal(self):
         return self.ratioString
 
+    def __eq__(self, other) -> bool:
+        '''
+        Establishes equality between two time signatures
+        on the basis of their having both
+        the same beatSequence (e.g., {{1/8+1/8}+{1/8+1/8+1/8}})
+        and the same ratioString (e.g., '2/8+3/8').
+
+        We need the two-part comparison for
+        pulling apart equality among a range of metrical structures.
+
+        For example, two 5-time meters with a different
+        beat structure count as different metres
+        despite the fact that they could both be written as '5/8'.
+        As such, this is a more restrictive test than
+        :meth:`~music21.meter.TimeSignature.ratioEqual`
+        which would return True for all '5/8's.
+
+        >>> oneKindOf5 = meter.TimeSignature('2+3/8')
+        >>> oneKindOf5.ratioString
+        '2/8+3/8'
+
+        >>> sameKindOf5 = meter.TimeSignature('2+3/8')
+        >>> sameKindOf5.ratioString
+        '2/8+3/8'
+
+        >>> oneKindOf5.__eq__(sameKindOf5)
+        True
+
+        >>> otherKindOf5 = meter.TimeSignature('3+2/8')
+        >>> otherKindOf5.ratioString
+        '3/8+2/8'
+
+        >>> oneKindOf5.__eq__(otherKindOf5)
+        False
+
+        Now for a complementary test the other way round:
+        two meters with the same internal structure
+        but a different written time signature.
+
+        >>> oneKindOf44 = meter.TimeSignature('4/4')
+        >>> sameKindOf44 = meter.TimeSignature()
+        >>> oneKindOf44 == sameKindOf44
+        True
+
+        >>> otherKindOf44 = meter.TimeSignature('Cut')
+        >>> oneKindOf44 == otherKindOf44
+        False
+
```

Comment:
Here are some things that weren't tested:

Is TS('Cut') == TS('2/2')  (I think no)

Is TS('slow 6/8') == TS('fast 6/8') ?  (I think no, and beatSequence will take care of it?)

BIG ONE:

Is TS('4/4') == note.Note() and reversed: is note.Note == TS('4/4')

you're find that your test will raise an attribute error.

Look at the __eq__ for other music21.note -- and others to fix this.  The first line should always be something like:

```
if not isinstance(other, type(self)):
    return False  # or return NotImplemented -- we do both in the current code, should unify
```

Reply:
Thanks @mscuthbert , I'll implement these tests to keep the code up-to-date with the discussion ... though I stopped adding tests before because even the few existing tests I already don't pass. Apologies if that wasn't clear.

There's something spooky going on here with `meter.TimeSignature` and ... and we're bit early for Halloween.

Shall we try and fix-finish this off as a placeholder to be superseded by the higher level #1458 at some point? I.e., as a kind of local test case before the more ambitious project. Or do you prefer to implement that higher level solution first?